plugins/SlicerT/SlicerTWaveform.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * SlicerTWaveform.h - declaration of class SlicerTWaveform
+ *
+ * Copyright (c) 2023 Daniel Kauss Serna <daniel.kauss.serna@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_SlicerT_Waveform_H
+#define LMMS_SlicerT_Waveform_H
+
+#include <QApplication>
+#include <QFontMetrics>
+#include <QInputDialog>
+#include <QMouseEvent>
+#include <QPainter>
+
+#include "Instrument.h"
+#include "SampleBuffer.h"
+
+namespace lmms {
+
+class SlicerT;
+
+namespace gui {
+
+class SlicerTWaveform : public QWidget
+{
+	Q_OBJECT
+
+public slots:
+	void updateUI();
+	void isPlaying(float current, float start, float end);
+
+public:
+	SlicerTWaveform(int w, int h, SlicerT* instrument, QWidget* parent);
+
+	// predefined sizes
+	const int m_seekerHorMargin = 5;
+	const int m_seekerHeight = 38; // used to calcualte all vertical sizes
+	const int m_middleMargin = 6;
```

Comment:
```suggestion
	static constexpr int s_seekerHorMargin = 5;
	static constexpr int s_seekerHeight = 38; // used to calculate all vertical sizes
	static constexpr int s_middleMargin = 6;
```
Constants can be made `static constexpr`. I changed the names so these would need to be fixed where they are used.

Reply:
Done